Show queuelen and queuemax

ton31337 opened the pull request:
Show queuelen and queuemax. These two values would be very cool in addition to current queuepct, because for monitoring. Something like: https://raw.githubusercontent.com/ton31337/tools/master/backlog_sysdig_all.lua
@ldegio @gianlucaborello any thoughts?

gianlucaborello commented:
Thanks.

The main issue I see with this is that you modified an event which was already set in stone. We don't change parameters for events that have already been out in an official release, otherwise we break compatibility with older trace files. In this case, if you run your branch on a 0.1.102 trace file you get:

```
$ ./sysdig -r ~/trace.scap evt.type=accept
3471 11:42:14.633164175 3 nginx (2171) > accept flags=0
sysdig: /home/gianluca/src/sysdig/userspace/libsinsp/event.cpp:1338: const char* sinsp_evt::get_param_as_str(uint32_t, const char**, sinsp_evt::param_fmt): Assertion `payload_len == sizeof(uint32_t)' failed.
Aborted
```

The "dirty" solution we've used so far is to duplicate events, such as https://github.com/draios/sysdig/blob/dev/driver/event_table.c#L256 and https://github.com/draios/sysdig/blob/dev/driver/event_table.c#L240

ton31337 commented:
Got it, will try to make some changes regarding your notes :+1:

ton31337 commented:
@gianlucaborello do you mean I should add something like this?:
```
/* PPME_SOCKET_ACCEPT_E */{"accept", EC_NET, (enum ppm_event_flags)(EF_CREATES_FD | EF_MODIFIES_STATE | EF_OLD_VERSION), 0},
/* PPME_SOCKET_ACCEPT_X */{"accept", EC_NET, (enum ppm_event_flags)(EF_CREATES_FD | EF_MODIFIES_STATE | EF_OLD_VERSION), 3, {{"fd", PT_FD, PF_DEC}, {"tuple", PT_SOCKTUPLE, PF_NA}, {"queuepct", PT_UINT8, PF_DEC} } },
/* PPME_SOCKET_ACCEPT_X */{"accept", EC_NET, (enum ppm_event_flags)(EF_CREATES_FD | EF_MODIFIES_STATE | EF_OLD_VERSION), 5, {{"fd", PT_FD, PF_DEC}, {"tuple", PT_SOCKTUPLE, PF_NA}, {"queuepct", PT_UINT8, PF_DEC}, {"queuelen", PT_UINT32, PF_DEC}, {"queuemax", PT_UINT32, PF_DEC} } },
```
compiles without any errors, but seems it needs something more here, because doesn't work. How it knows about 'duplicate'?

ton31337 commented:
@gianlucaborello how should I do with `./driver/ppm_events_public.h` ? I see there is enumeration with E and X. Should I create the same pair E/X or can I use only Y and define it there?

ton31337 commented:
@gianlucaborello what is missing in latest commit?

gianlucaborello commented:
Well this is somehow worse than before :)

1) The new events definition must be placed at the end of the list, so that it follows the same enum order. With this change essentially, all events after `accept()` are shifted and so old trace files are completely broken, and probably a good portion of live too

2) Seems like there are still some references to the old events in the `parse_socketcall()` function

3) In filterchecks.cpp's switch, why did you handle `PPME_SOCKET_ACCEPT4_5_E` in a different way? Same thing for the switch in parsers.cpp for the `PPME_SOCKET_ACCEPT4_5_X` event

4) The replaced events should be marked with the `EF_OLD_VERSION` flag

ton31337 commented:
1) Yea, that was a key with enumeration order.. Looks better after reordering ;-)
2) Fixed
3) Fixed
4) Added

Looks much better now, no kernel panics, and tests are :+1: I'm beginning to understand how it works eventually :laughing:

gianlucaborello commented:
Thanks, much better now.

In filterchecks.cpp's `switch`, there's still an indentation issue, which also shows up in the diff from github.

ton31337 commented:
Fixed :)

gianlucaborello commented:
Merged, thanks for this

ton31337 commented:
:bow: by the way @gianlucaborello how often do you guys release the new version? Is here any regular release cycle or something like that?

gianlucaborello commented:
Not regular cycles, we release it when it's ripe enough :)

There might be a release later this week including everything from the dev branch.
